README.md

```diff
@@ -151,21 +144,113 @@ domain_beanPropertyValue1_key1_key2_...keyN_attrName{beanpropertyName2="beanProp
 ```
 If a given part isn't set, it'll be excluded.
 
-## Integration Testing
+## HTTP Authentication (optional)
+
+HTTP BASIC authentication it supported using various configuration formats.
+
+---
+
+Simple example (plaintext configuration value):
+
+```yaml
+httpServer:
+  authentication:
+    basic:
+      username: Prometheus
+      password: secret
+```
+
+---
+
+More secure example (SHA-256 using a salted password `SHA256(<salt>:<password>)`):
+
+```yaml
+httpServer:
+  authentication:
+    basic:
+      username: Prometheus
+      hash: 2bf7ed4906ac065bde39f7508d6102a6cdd7153a929ea883ff6cd04442772c99
+      algorithm: SHA-256
+      salt: U9i%=N+m]#i9yvUV:bA/3n4X9JdPXf=n
+```
+
+**Notes**
+
+- `algorithm` type must match a supported Java algorithm (JVM specific)
+
+- `salt` is random string that you choose
+
+---
+
+Secure example (PBKDF2WithHmacSHA1):
+
+```yaml
+httpServer:
+  authentication:
+    basic:
+      username: Prometheus
+      password: B6:9C:5C:8A:10:3E:41:7B:BA:18:FC:E1:F2:0C:BC:D9:65:70:D3:53:AB:97:EE:2F:3F:A8:88:AF:43:EA:E6:D7:FB:70:14:23:F9:51:29:5C:3A:9F:65:C3:20:EE:09:C9:C6:8A:B7:D3:0A:E1:F3:10:2B:9B:36:3F:1F:B6:1D:52:A7:9C:CB:AD:55:25:46:C5:73:09:6C:38:9C:F2:FD:82:7F:90:E5:31:EF:7E:3E:6B:B2:0C:38:77:23:EC:3A:CF:29:F7:E5:4D:4E:CC:35:7A:C2:E5:CB:E3:B3:E5:09:2B:CC:B9:40:26:A4:28:E9:5F:2D:18:B2:14:41:E7:4D:5B
```

Comment:
The name `password` is outdated.

Apart from that: In the `SHA-256` example above the hash was a lowercase hex string. Here it's an uppercase hex string with colons. I think it would be more readable to use a lowercase hex string everywhere. To remove the colons and convert to lowercase in the command line you can pipe the openssl output through `tr` like this:
```
openssl ... | tr -d : | tr '[:upper:]' '[:lower:]
```

Reply:
Refactored code to support...

- uppercase with `:`" separators (`openssl` standard output)
- lowercase with `:` separators
- uppercase without `:` separators
- lowercase without `:` separators